spec/tapioca/generators/base_generator_spec.rb

```diff
@@ -0,0 +1,170 @@
+# typed: true
+# frozen_string_literal: true
+
+require "spec_helper"
+require "generator_spec"
+require "tapioca/generators/base_generator"
+
+module Tapioca
+  module Generators
+    class BaseGeneratorSpec < GeneratorSpec
+      describe "#rbi_header" do
+        it "displays when config.file_header is true" do
+          config = ConfigBuilder.from_options(:dsl, {})
+          generator = Tapioca::Generators::BaseGenerator.new(config)
+          result = generator.rbi_header("bin/tapioca dsl")
+          expected = <<~RUBY
+            # DO NOT EDIT MANUALLY
+            # This is an autogenerated file for .
+            # Please instead update this file by running `bin/tapioca dsl`.
+
+          RUBY
+          assert_equal(expected, result)
+        end
+
+        it "doesn't display when config.file_header is false" do
+          config = ConfigBuilder.from_options(:dsl, { "file_header" => false })
```

Comment:
I think the `config` being repeated in each test case here is a symptom of the bad separation of concerns it brings when required to build a `Generator`.